Sources/App/Commands/Ingestor.swift

```diff
@@ -26,40 +26,43 @@ struct IngestorCommand: Command {
 
 
 func ingest(client: Client, database: Database, limit: Int) -> EventLoopFuture<Void> {
-    fetchMetadata(client: client, database: database, limit: limit)
-        .flatMapEachThrowing { result in
-            do {
-                let (pkg, md) = try result.get()
-                return try insertOrUpdateRepository(on: database, for: pkg, metadata: md)
-                    .flatMap {
-                        pkg.status = .ok
-                        return pkg.save(on: database)
-                    }
-            } catch {
-                return recordIngestionError(database: database, error: error)
-            }
-        }
+    Package.fetchUpdateCandidates(database, limit: limit)
+        .flatMapEach(on: database.eventLoop) { fetchMetadata(for: $0, with: client) }
+        .flatMapEachThrowing { updateTables(on: database, result: $0) }
```

Comment:
As discussed, I pulled metadata fetching out into the top level `ingest` command and also broke out the table updates into their own method. Reads nicer now, I think.